packages/hardhat-chai-matchers/src/bigNumber.ts

```diff
@@ -134,6 +136,33 @@ function overwriteBigNumberFunction(
         expected,
         actual
       );
+    } else if (functionName === "eq" && chaiUtils.flag(this, "deep")) {
```

Comment:
The problem is that the inputs to `.to.deep.equal` will be arrays/objects, so they will never be big numbers. I guess we could traverse all objects and check if any member is a big number, but that doesn't seem like a good idea to me.

So yes, we are changing the behavior of _all_ `.to.deep.equal` calls. It would be nice if deep equality would just work when you extend the basic `.equal` matchers, but sadly that's not how chai works. That's why I re-implemented the deep equality comparison, using the same package chai uses, but using a custom comparator function. I'm not super happy with this solution, but I can't think of a better approach.

Reply:
I see, that makes sense to me, thank you for the explanation.